lib/spack/spack/util/path.py

```diff
@@ -122,7 +122,7 @@ def path_to_os_path(*pths):
     """
     ret_pths = []
     for pth in pths:
-        if type(pth) is str and not is_path_url(pth):
+        if str in type(pth).__mro__ and not is_path_url(pth):
```

Comment:
Would `isinstance` work as well? I think that would be more straightforward if so.

Reply:
I believe it would, I'll change that.